pkg/common/secret/manager.go

```diff
@@ -0,0 +1,34 @@
+package secret
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/observability-operator/pkg/monitoring"
+)
+
+type Manager interface {
+	GenerateGenericSecret(secretName string, secretNamespace string, key string, value string) (*corev1.Secret, error)
+}
+
+type SimpleManager struct {
+}
+
+func (m SimpleManager) GenerateGenericSecret(secretName string, secretNamespace string,
```

Comment:
Do we need a manager for that? A simple function should have been enought right?

Reply:
True, but I created a manager because I thought that this might prove useful in the future if we need to add other methods there

Reply:
Can we create a manager later if we need to? This feels like premature optimization that is not really helping here imo

Reply:
Actually I found an additional usage for this manager : a `DeleteSecret` function which is really useful for deleting both mimir secrets as well as remotewrite secrets.

Reply:
Why is a util function not enough ? Does it requires you to create an object and pass it around?

Reply:
Hmmm, maybe using this file as a list of utils functions might be enough indeed.

Reply:
Done

Reply:
I guess it's not pushed yet :D